### PR TITLE
Support pagination in AsyncReferenceClient.get_stock_financials_vx

### PR DESCRIPTION
**Pull request by MuradGitHub**
Added support for pagination for AsyncReferenceClient.get_stock_financials_vx using standard approach applied in other methods of the AsynRerrenceClient

### CODE REVIEW

**Comment by MuradGitHub** [comment]
A pull request combining both pagination updates to get_stock_financials and get_conditions

File: polygon/reference_apis/reference_api.py [hunk 1]
```diff
@@ -1428,7 +1447,9 @@ async def get_stock_financials_vx(self, ticker: str = None, cik: str = None, com
                                       period_of_report_date_lt=None, period_of_report_date_lte=None,
                                       period_of_report_date_gt=None, period_of_report_date_gte=None,
                                       time_frame=None, include_sources: bool = False, order='asc',
-                                      limit: int = 50, sort='filing_date', raw_response: bool = False):
+                                      limit: int = 50, sort='filing_date', all_pages: bool = False,
+                                      max_pages: int = None, merge_all_pages: bool = True, verbose: bool = False,
```
Comment (pssolanki111) [inline polygon/reference_apis/reference_api.py:1451]: please cover the non-async version of this method.

**Review by pssolanki111 — CHANGES_REQUESTED** [review]

**Comment by MuradGitHub** [comment]
Feedback addressed + added two level warnings for StocksClient..get_aggregate_bars function

File: polygon/stocks/stocks.py [hunk 2]
```diff
@@ -358,7 +358,7 @@ def get_daily_open_close(self, symbol: str, date, adjusted: bool = True,
     def get_aggregate_bars(self, symbol: str, from_date, to_date, adjusted: bool = True,
                            sort='asc', limit: int = 5000, multiplier: int = 1, timespan='day', full_range: bool = False,
                            run_parallel: bool = True, max_concurrent_workers: int = cpu_count() * 5,
-                           warnings: bool = True, high_volatility: bool = False, raw_response: bool = False):
+                           warnings: bool = True, info: bool = True, high_volatility: bool = False, raw_response: bool = False):
```
Comment (pssolanki111) [inline polygon/stocks/stocks.py:361]: the only small concern here is that this base method is used by other clients as well. While it won't result in any errors (due to using named default attribute), it changes the base method signature which isn't a good practice.

We can either 
- make this top level change for the other 3 clients (forex, crypto, options)
- OR skip this in this PR. I'll make the change in the next release

**Review by pssolanki111 — COMMENTED** [review]
minor question. lemme what you'd prefer to go with. 
Rest of the PR is good to go

File: polygon/stocks/stocks.py (hunk 2 above)
Comment (MuradGitHub) [reply polygon/stocks/stocks.py:361]: Makes sense to sync the other clients with the changed API. I'll update the PR in the next couple of days

**Comment by MuradGitHub** [comment]
Updated the PR as requested. in addition, I added wrapper for requests.Response.json as I see there are some responses that the json method can't parse / throws an exception on which crashes the script. The wrapper just allows the client to report this as a failure to source data for a particular ticker

**Review by pssolanki111 — APPROVED** [review]
Merging. Will make some changes and release to PyPi. 

Thanks a lot for the contribution @MuradGitHub

**Pull request merged**